### PR TITLE
Added Entities and Updated UserDataAccessObject

### PR DESCRIPTION
**Pull request by HussainE4**
Added classes and interfaces for our main entities (User, Track, Playlist, Artist) based on the proposed entities from our proposal. Up till now, each class only has getter methods. Also updated the User DataAccessObject (will definitely need another one for the spotify stuff). Instead of storing usernames and passwords, we now keep usernames and users. 

NOTE: We are not actually storing the data anywhere, it is just added to the dictionary. Need to add data persistence with json files

### CODE REVIEW

File: src/app/Main.java [hunk 1]
```diff
@@ -20,6 +22,8 @@ public static void main(String[] args) {
         ViewManager viewManager = new ViewManager(viewManagerModel, views, cardLayout);
 
 
+        // Create a new DAO in MAIN that should be passed to the Views
+        UserDataAccessObject UserDataAccessObject = new UserDataAccessObject(new UserFactory());
```
Comment (skrukwa) [inline src/app/Main.java:26]: views do not interact with DAOs

only use case interactors interact with DAOs

File: src/data_access/UserDataAccessObject.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import entities.UserFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserDataAccessObject {
+
+    private final Map<String, User> users;
+
+    private UserFactory userFactory;
```
Comment (skrukwa) [inline src/data_access/UserDataAccessObject.java:13]: should be final?

File: src/data_access/UserDataAccessObject.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        User fake_user1 = userFactory.create("_Jason", "pass123");
+        User fake_user2 = userFactory.create("bob445", "bananaPASS");
+        users.put(fake_user1.getName(), fake_user1);
+        users.put(fake_user2.getName(), fake_user2);
+
+        // SOMETHING WE NEED TO FIGURE OUT: HOW WE'RE STORING THE USERS AND OTHER DATA PERSISTENCE
+    }
+
+
+    public User getUser(String username){return users.get(username);}
```
Comment (skrukwa) [inline src/data_access/UserDataAccessObject.java:27]: does not follow Google Java Style Guide 4.1.2

File: src/entities/CommonUser.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class CommonUser implements User{
```
Comment (skrukwa) [inline src/entities/CommonUser.java:5]: Google Java Style Guide 4.6.2

File: src/entities/CommonUser.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    private final String name;
```
Comment (skrukwa) [inline src/entities/CommonUser.java:6]: should use "username" and "getUsername", not to be confused with a users given name or government name

File: src/entities/Track.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    String getID();
+
+    String getImageURL();
+
+    String getName();
+
+    String getArtists();
+
+    int getDuration();
+
```
Comment (skrukwa) [inline src/entities/Track.java:14]: add "isExplicit()" method

File: src/entities/Track.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
```
Comment (skrukwa) [inline src/entities/Track.java:15]: 4.6.1 Google Java Style Guide

File: src/entities/CommonTrack.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
+package entities;
+
+public class CommonTrack implements Track{
+    String id;
+    String imageURL;
+    String name;
+    String artists;
+    int duration;
+
```
Comment (skrukwa) [inline src/entities/CommonTrack.java:9]: fields should all be final

Comment (skrukwa) [inline src/entities/CommonTrack.java:9]: add "final boolean explicit"

File: src/entities/CommonPlaylist.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class CommonPlaylist implements Playlist{
+
+    private final String name;
+    private final User user;
```
Comment (skrukwa) [inline src/entities/CommonPlaylist.java:8]: use more descriptive (SCREAMING) "owner" field instead of "user"

**Review by skrukwa — CHANGES_REQUESTED** [review]
see comments, and ideally submit PRs using the template (even if all boxes are not check marked)

File: src/app/Main.java (hunk 1 above)
Comment (HussainE4) [reply src/app/Main.java:26]: Correction: This is given to the UseCaseFactory which uses it when creating Controllers

**Review by skrukwa — CHANGES_REQUESTED** [review]
you still have comments from my first request that were not resolved. also please change  getExplicit to isExplicit for Track

**Review by skrukwa — APPROVED** [review]

**Pull request merged**